### PR TITLE
Habib/404 from binary explore deriv

### PR DESCRIPTION
**Pull request by ShuvoHabib**
Changes:

Issue: 404 error message is displayed when user tries to switch the language after navigating to Deriv.com from Binary.com 

Fix: Add Regex fix for language urls ending without '/'

## Type of change

-   [x] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/components/localization/language-switcher.js [hunk 1]
```diff
@@ -41,14 +41,14 @@ const LanguageSwitch = ({ i18n, is_high_nav, short_name }) => {
...
-
+        
         if (`/${current_lang}/` !== id) {
             const current_path = window.location.pathname
             const current_hash = window.location.hash
             const destination_path = `${path}${
                 current_lang === 'en'
                     ? current_path.replace(/\//u, '')
-                    : current_path.replace(/\/.+?\//u, '')
+                    : current_path.replace(/\/.+?(\/)|(\/\w+)/u, '')
```
Comment (suthesan-deriv) [inline src/components/localization/language-switcher.js:51]: Can we have a comment here to explain what the regex does please 🙏

Comment (mustofa-binary) [inline src/components/localization/language-switcher.js:51]: can we make this a function, normalizeURL in utils 🙏

File: src/components/localization/language-switcher.js [hunk 2]
```diff
@@ -41,14 +41,14 @@ const LanguageSwitch = ({ i18n, is_high_nav, short_name }) => {
         const { id } = e.target
         const current_lang = language || 'en'
         const path = id === '/en/' ? '/' : id
-
+
```
Comment (mustofa-binary) [inline src/components/localization/language-switcher.js:44]: nit: extra space

File: src/components/localization/language-switcher.js (hunk 1 above)
Comment (ShuvoHabib) [reply src/components/localization/language-switcher.js]: Resolved, thank you

File: src/components/localization/language-switcher.js (hunk 2 above)
Comment (ShuvoHabib) [reply src/components/localization/language-switcher.js]: fixed this weird space

File: src/components/localization/language-switcher.js (hunk 1 above)
Comment (ShuvoHabib) [reply src/components/localization/language-switcher.js]: Created a variable with a self explanatory name.
const path_with_or_without_slash = /\/.+?(\/)|(\/\w+)/u

Comment (ShuvoHabib) [reply src/components/localization/language-switcher.js]: Created a variable with a self explanatory name.
const path_with_or_without_slash = /\/.+?(\/)|(\/\w+)/u

**Review by sean-binary — APPROVED** [review]

**Review by sean-binary — APPROVED** [review]

**Pull request merged**